### ISSUE TITLE
Read timeout during VM delete

### ISSUE THREAD

**Issue by dsboulder** [cloudfoundry-incubator/bosh-azure-cpi-release#203]
I got this error while deleting a bunch of VMs:
```
   Failed deleting instances > consul_server/0 (d6ac3c7e-a47e-4820-ad16-f35e67ebaf6e): http_get_response - #<Net::ReadTimeout: Net::ReadTimeout>
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/protocol.rb:158:in `rescue in rbuf_fill'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/protocol.rb:152:in `rbuf_fill'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/protocol.rb:134:in `readuntil'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/protocol.rb:144:in `readline'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/http/response.rb:39:in `read_status_line'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/http/response.rb:28:in `read_new'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/http.rb:1414:in `block in transport_request'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/http.rb:1411:in `catch'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/http.rb:1411:in `transport_request'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/http.rb:1384:in `request'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/http.rb:1377:in `block in request'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/http.rb:853:in `start'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/http.rb:1375:in `request'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb:988:in `http_get_response'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb:1226:in `http_post'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb:858:in `get_storage_account_keys_by_name'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/blob_manager.rb:289:in `block in initialize_blob_client'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/blob_manager.rb:286:in `synchronize'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/blob_manager.rb:286:in `initialize_blob_client'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/blob_manager.rb:100:in `get_blob_properties'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/disk_manager.rb:83:in `has_disk?'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/disk_manager.rb:30:in `delete_disk'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/vm_manager.rb:83:in `delete'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/cloud.rb:147:in `block in delete_vm'
/var/vcap/packages/bosh_azure_cpi/vendor/bundle/ruby/2.2.0/gems/bosh_common-1.3262.4.0/lib/common/thread_formatter.rb:49:in `with_thread_name'
/var/vcap/packages/bosh_azure_cpi/lib/cloud/azure/cloud.rb:145:in `delete_vm'
/var/vcap/packages/bosh_azure_cpi/vendor/bundle/ruby/2.2.0/gems/bosh_cpi-1.3262.4.0/lib/bosh/cpi/cli.rb:70:in `public_send'
/var/vcap/packages/bosh_azure_cpi/vendor/bundle/ruby/2.2.0/gems/bosh_cpi-1.3262.4.0/lib/bosh/cpi/cli.rb:70:in `run'
/var/vcap/packages/bosh_azure_cpi/bin/azure_cpi:35:in `<main>' (00:03:52)
```

Here's the part of the debug log related:
```
I, [2016-09-08T17:10:16.198014 #7844]  INFO -- : http_post - 0: trying to post https://management.azure.com//subscriptions/b269df04-a1d5-4e2b-9368-d4abf33bc411/resourceGroups/dso
msept6-pcf-resource-group/providers/Microsoft.Storage/storageAccounts/dsomsept6ompremium/listKeys?api-version=2015-06-15
D, [2016-09-08T17:10:16.198173 #7844] DEBUG -- : http_get_response - 0: #<Net::HTTP::Post POST>, URI: https://management.azure.com//subscriptions/b269df04-a1d5-4e2b-9368-d4abf33b
c411/resourceGroups/dsomsept6-pcf-resource-group/providers/Microsoft.Storage/storageAccounts/dsomsept6ompremium/listKeys?api-version=2015-06-15
opening connection to management.azure.com:443...
opened
starting SSL for management.azure.com:443...
SSL established
<- "POST //subscriptions/b269df04-a1d5-4e2b-9368-d4abf33bc411/resourceGroups/dsomsept6-pcf-resource-group/providers/Microsoft.Storage/storageAccounts/dsomsept6ompremium/listKeys?
api-version=2015-06-15 HTTP/1.1\r\nAccept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3\r\nAccept: */*\r\nUser-Agent: BOSH-AZURE-CPI\r\nContent-Length: 0\r\nContent-Type: app
lication/json\r\nAuthorization: Bearer **REDACTED**\r\nConnection: close\r\nHost: management.azure.com\r\n\r\n"
<- ""
Conn close because of error Net::ReadTimeout
Conn close because of error Net::ReadTimeout
E, [2016-09-08T17:11:17.287856 #7844] ERROR -- : http_get_response - #<Net::ReadTimeout: Net::ReadTimeout>
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/protocol.rb:158:in `rescue in rbuf_fill'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/protocol.rb:152:in `rbuf_fill'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/protocol.rb:134:in `readuntil'
/var/vcap/packages/ruby_azure_cpi/lib/ruby/2.2.0/net/protocol.rb:144:in `readline'
```

**Comment by AbelHu** [comment]
@dsboulder Thank you for reporting this issue. We will,
1. Ask Azure CAT team to find the root cause for this error
2. Handle ReadTimeout in CPI as same as OpenTimeout

**Comment by AbelHu** [comment]
@dsboulder Code has been submitted in this [commit](https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/commit/d3e3e1dc993bcf52906dc3bf01756a7bb8b25ff7)

**Comment by AbelHu** [comment]
Close it since azure-storage-ruby has fixed this in v0.11.1 and Azure CPI has upgraded the SDK to v0.11.1 in this [commit](https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/commit/90be7c11f2ede00136e177873a651e96d018c981).
